adversary/repeat.go

```diff
@@ -0,0 +1,64 @@
+package adversary
+
+import (
+	"github.com/filecoin-project/go-f3/gpbft"
+)
+
+type (
+	Repeat struct {
+		id            gpbft.ActorID
+		host          gpbft.Host
+		echoCountDist CountSampler
+	}
+	CountSampler interface {
+		// TODO replace with math/rand/v2 rand.Source once upgraded to go 1.22.
+		Uint64() uint64
+	}
+)
+
+func NewRepeat(id gpbft.ActorID, host gpbft.Host, echoCountDist CountSampler) *Repeat {
```

Comment:
Please add a comment explaining the intent of this adversary. There is some nuance to capture, e.g. that the messages are re-signed by this participant (it's not implementing a network-level duplication). I would say the messages are copied or echo'd by this participant, rather than repeated, though the distinction is fine.